src/include/ipxe/efi/ProcessorBind.h

```diff
@@ -10,20 +10,54 @@ FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL );
  *  - mcb30
  */
 
-#if __i386__
+#ifdef EFI_HOSTONLY
+
+/*
+ * We cannot rely on the EDK2 ProcessorBind.h headers when compiling a
+ * binary for execution on the build host itself, since the host's CPU
+ * architecture may not even be supported by EDK2.
+ */
+
+/* Define the basic integer types in terms of the host's <stdint.h> */
+#include <stdint.h>
+typedef int8_t INT8;
+typedef int16_t INT16;
+typedef int32_t INT32;
+typedef int64_t INT64;
+typedef uint8_t UINT8;
+typedef long INTN;
+typedef uint16_t UINT16;
+typedef uint32_t UINT32;
+typedef uint64_t UINT64;
+typedef unsigned long UINTN;
+typedef int8_t CHAR8;
+typedef int16_t CHAR16;
+typedef uint8_t BOOLEAN;
+
+/* Define EFIAPI as whatever API the host uses by default */
+#define EFIAPI
+
+/* Define an architecture-neutral MDE_CPU macro to prevent build errors */
+#define MDE_CPU_EBC
+
+#else /* EFI_HOSTONLY */
+
+#ifdef __i386__
 #include <ipxe/efi/Ia32/ProcessorBind.h>
 #endif
 
-#if __x86_64__
+#ifdef __x86_64__
 #include <ipxe/efi/X64/ProcessorBind.h>
 #endif
 
-#if __arm__
+#ifdef __arm__
```

Comment:
I think that the #if was intentional here, there used to be an Apple compiler that defined one of these to 0 for an unsupported architecture.

May not be very relevant anymore today.

Reply:
We have several other points within the codebase that use `#if defined(__arm__)`, so this change can't introduce any additional problems.  I changed it here so that there was a consistent use of `#ifdef` within `ProcessorBind.h` rather than a mix of `#ifdef` and `#if`.

(The alternative would have been to use an explicit `#define EFI_HOSTONLY 1` instead of just `#define EFI_HOSTONLY` in `elf2efi.c`, but that would itself be inconsistent with the way that `#define` flags are generally used in iPXE.)